### PR TITLE
code cleanup: compilation options

### PR DESCRIPTION
**Pull request by djphoenix**
Fixes: #1549

A bunch of `#ifdef` stuff cleanups, including:

- Check header presence with `__has_include` instead of autoconf's `HAVE_XXX` macro
- Standartization of defines: consistent naming, define to 1 and check with `#if` instead of `#ifdef`
- Compile MBROLA-related code only if configured `--with-mbrola`
- Compile tests with flags that match configuration (including async, klatt, speechplayer, mbrola)
- Fix for API test when compiled with async

### CODE REVIEW

File: src/libespeak-ng/event.c [hunk 1]
```diff
@@ -45,8 +45,8 @@ static pthread_cond_t my_cond_stop_is_acknowledged;
 static bool my_stop_is_acknowledged = false;
 static bool my_terminate_is_required = 0;
 // my_thread: polls the audio duration and compares it to the duration of the first event.
-static pthread_t my_thread;
-static bool thread_inited;
+static pthread_t my_thread = 0;
```
Comment (sthibaul) [inline src/libespeak-ng/event.c:48]: No, the type of `pthread_t` is not specified, one cannot assume that it can be set to 0 (and not even assume that 0 is a non-thread-id value)

File: src/libespeak-ng/fifo.c [hunk 2]
```diff
@@ -131,6 +131,9 @@ espeak_ng_STATUS fifo_add_command(t_espeak_command *the_command)
 espeak_ng_STATUS fifo_add_commands(t_espeak_command *command1, t_espeak_command *command2)
 {
 	espeak_ng_STATUS status;
+	if (!my_thread) {
+		return ENS_NOT_INITIALIZED;
+	}
```
Comment (sthibaul) [inline src/libespeak-ng/fifo.c:136]: This will break on systems where pthread_t can be 0 for some threads.

Comment (djphoenix) [reply src/libespeak-ng/fifo.c:136]: I don't know any platform where pthread_t may be zero. But OK, I will rewrite this on another `bool` variable.

File: src/libespeak-ng/speech.c [hunk 3]
```diff
@@ -118,17 +118,14 @@ static int dispatch_audio(short *outbuf, int length, espeak_EVENT *event)
...
 			if (out_samplerate != voice_samplerate) {
-#ifdef HAVE_PCAUDIOLIB_AUDIO_H
+#if USE_LIBPCAUDIO
 				if (out_samplerate != 0) {
 					// sound was previously open with a different sample rate
 					audio_object_close(my_audio);
 					out_samplerate = 0;
-#ifdef HAVE_SLEEP
-					sleep(1);
-#endif
```
Comment (sthibaul) [inline src/libespeak-ng/speech.c:128]: I'd say make this a separate commit, we don't know what impact it may have

Comment (sthibaul) [reply src/libespeak-ng/speech.c:128]: Ah, configure actually wasn't looking for the sleep function, so this was indeed dead, ok.

**Comment by sthibaul** [comment]
I haven't checked details, but this looks good to me overall.

File: src/libespeak-ng/speech.c (hunk 3 above)
Comment (djphoenix) [reply src/libespeak-ng/speech.c:128]: HAVE_SLEEP does not defined from any sources (including autoconf). It is indeed dead code.

**Comment by djphoenix** [comment]
Oops, I have found that check for async enablement should be moved to InitializeOutput to reflect actual `my_mode` value.

**Review by alex19EP — APPROVED** [review]
